tests/test_metadata.py

```diff
@@ -198,13 +198,25 @@ def test_append_git_log_to_metadata(source_tree):
     # Match 2 log lines.
     assert re.match(
         r"[0-9A-Fa-f]+\ncode/c\n+[0-9A-Fa-f]+\ncode/b\n+",
-        mdata.sources[0].git.log,
+        mdata.sources[2].git.log,
         re.MULTILINE,
     )
     # Make sure the local path field is not recorded.
     assert not mdata.sources[0].git.local_path is None
 
 
+def test_synthtool_and_cwd_git_sources_in_metadata(source_tree):
+    with metadata.MetadataTrackerAndWriter(source_tree.tmpdir / "synth.metadata"):
```

Comment:
Can you add a comment about why this is starting and closing a context without seeming to do anything? It looks like you're using it to create the synth.metadata file, but it's hard to understand at first glance.